### PR TITLE
Issue #260: Redirect to user/login when anonymous views user page.

### PR DESCRIPTION
**Pull request by quicksketch**
Fixes backdrop/backdrop-issues#260

Replaces https://github.com/backdrop/backdrop/pull/752.

### CODE REVIEW

File: core/includes/theme.maintenance.inc [hunk 1]
```diff
@@ -93,6 +93,7 @@ function _backdrop_maintenance_theme() {
   backdrop_add_css($path . '/css/system.css');
   backdrop_add_css($path . '/css/system.admin.css');
   backdrop_add_css($path . '/css/system.theme.css');
+  backdrop_add_css($path . '/css/messages.theme.css', array('group' => CSS_SYSTEM, 'every_page' => TRUE));
```
Comment (opi) [inline core/includes/theme.maintenance.inc]: Unrelated change, this PR should be rebased ;)

**Review by opi — CHANGES_REQUESTED** [review]

**Comment by quicksketch** [comment]
Thanks @opi! Rerolled to remove that unrelated change to theme.maintenance.inc and also fixed the remaining failing test.

**Review by opi — APPROVED** [review]
tested, works perfectly as expected.

**Comment by serundeputy** [comment]
Thanks @opi @jenlampton and @quicksketch !

Works as expected merged into `1.x` for `1.9.0.`!

**Pull request merged**